tidy3d/components/eme/simulation.py

```diff
@@ -582,13 +620,26 @@ def _validate_sweep_spec(self):
                     f"'num_modes={self.sweep_spec.num_modes}'; the maximum "
                     f"number of EME modes is '{self.max_num_modes}'."
                 )
-        if isinstance(self.sweep_spec, EMELengthSweep):
-            if len(self.sweep_spec.scale_factors) > MAX_NUM_SWEEP:
-                raise SetupError(
-                    "Simulation 'sweep_spec' has "
-                    f"'len(scale_factors)={len(self.sweep_spec.scale_factors)}', which exceeds "
-                    f"the maximum allowed '{MAX_NUM_SWEEP}'."
-                )
+        elif isinstance(self.sweep_spec, EMELengthSweep):
+            scale_factors_shape = self.sweep_spec.scale_factors.shape
+            if len(scale_factors_shape) == 2:
+                num_scale_factors = scale_factors_shape[1]
+                if num_scale_factors != self.eme_grid.num_cells:
+                    raise SetupError(
+                        "Simulation 'sweep_spec.scale_factors' has shape "
+                        f"'{scale_factors_shape}'. The size of the second dimension "
+                        "must equal the number of EME cells in the simulation, which is "
+                        f"'{self.eme_grid.num_cells}'."
+                    )
+        elif isinstance(self.sweep_spec, EMEFreqSweep):
+            for i, scale_factor in enumerate(self.sweep_spec.freq_scale_factors):
+                scaled_freqs = np.array(self.freqs) * scale_factor
+                if np.min(scaled_freqs) < MIN_FREQUENCY:
+                    raise ValidationError(
```

Comment:
reason for not using `SetupError`? If I recall, there might be some issues with using `ValidationError` handling with our new log / error handling.